proparse/src/main/java/org/prorefactor/proparse/antlr4/ProToken.java

```diff
@@ -67,15 +68,17 @@
 
   private String analyzeSuspend = "";
 
-  public ProToken(int type, String text) {
+  public ProToken(ABLNodeType type, String text) {
     this.type = type;
     this.channel = DEFAULT_CHANNEL;
     this.text = text;
     this.fileIndex = -1;
   }
 
   public ProToken(int type, int channel, int start, int stop, int line, int col) {
-    this.type = type;
+    this.type = ABLNodeType.getNodeType(type);
+    if (this.type == null)
+      throw new IllegalArgumentException("Invalid type number " + type);
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "Invalid type number " 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1192)